gossip/gossip.go

```diff
@@ -554,15 +564,16 @@ func (g *Gossip) bootstrap(stopper *stop.Stopper) {
 // is notified via the stalled conditional variable.
 func (g *Gossip) manage(stopper *stop.Stopper) {
 	stopper.RunWorker(func() {
-		// Loop until closed and there are no remaining outgoing connections.
 		for {
 			select {
 			case <-stopper.ShouldStop():
 				return
 			case c := <-g.disconnected:
 				g.doDisconnected(stopper, c)
-			case <-time.After(g.jitteredInterval(checkInterval)):
-				g.doCheckNetwork(stopper)
+			case nodeID := <-g.tighten:
+				g.tightenNetwork(stopper, nodeID)
+			case <-time.After(g.jitteredInterval(cullInterval)):
+				g.cullNetwork()
 			case <-time.After(g.jitteredInterval(stallInterval)):
```

Comment:
when it seems like something doesn't make sense, sometimes it just doesn't.. :-)
You're absolutely right, and your ticker example is the code that should be here.
Doesn't seem like we're exercising this code path in tests, so we should change
that.